plugin/storage/es/factory.go

```diff
@@ -239,8 +260,31 @@ var _ io.Closer = (*Factory)(nil)
 
 // Close closes the resources held by the factory
 func (f *Factory) Close() error {
+	var errs []error
+	for _, w := range f.watchers {
+		errs = append(errs, w.Close())
+	}
 	if cfg := f.Options.Get(archiveNamespace); cfg != nil {
-		cfg.TLS.Close()
+		errs = append(errs, cfg.TLS.Close())
+	}
+	errs = append(errs, f.Options.GetPrimary().TLS.Close())
+	return errors.Join(errs...)
+}
+
+func (f *Factory) onPrimaryPasswordChange() {
+	primaryClient, err := f.newClientFn(f.primaryConfig, f.logger, f.metricsFactory)
```

Comment:
You are using the old config here, not the new password. This needs a unit test

Reply:
The current tests and linters are fine now so I'm adding new tests now.

But could you help explain why password is not updated here? I thought when we recall f.newClientFn(f.primaryConfig, ...) then we'd recall [getConfigOptions](https://github.com/jaegertracing/jaeger/blob/main/pkg/es/config/config.go#L100) and get the latest password from password file path in config?

Reply:
This is why when you develop new functionality you often start with a unit test that verifies what you're trying to implement. In this case we want to observe that a new client is created once pwd file is changed - if you had that test you would've caught several bugs I already pointed out.

Reply:
Yeah sorry I'll add that test now^^ I have exams until Thursday so maybe I'll push the test after that.

Reply:
no need to apologize, your contributions are greatly appreciated.